distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Lazy.enso

```diff
@@ -0,0 +1,75 @@
+import project.Any.Any
+import project.Error.Error
+import project.Nothing.Nothing
+import project.Panic.Caught_Panic
+import project.Panic.Panic
+import project.Runtime.Ref.Ref
+
+## Holds a value that is computed on first access.
+type Lazy
+    ## PRIVATE
+    Lazy (cached_ref : Ref) (builder : Nothing -> Any)
+
+    ## PRIVATE
+    Eager (value : Any)
+
+    ## Creates a new lazy value.
+    new : Any -> Lazy
+    new ~lazy_computation =
+        builder _ = lazy_computation
```

Comment:
Just curious, is this dummy parameter necessary to avoid early evaluation of `lazy_computation`?

Reply:
Tbh I did not test it, but I'm pretty sure that fields are always stored 'eagerly' in an Atom - and the `builder` will be stored in an Atom a few lines later.

![image](https://user-images.githubusercontent.com/1436948/228277371-b311410d-72d6-4ee4-9a5b-5ea61b6ede5f.png)

So to store a 'thunk' I need to wrap it in a lambda.

Reply:
Fields will always be stored _"eagerly"_ in the `Atom` until we implement 
- #5085

Would you benefit from having _"lazy atom fields"_? Ask and we'll be more than glad to implement it!